### PR TITLE
Retired eg flag, eg-release, previous_release, eg_version and sub_dir…

### PR DESCRIPTION
**Pull request by thomasmaurel**
… from config files and associated modules. Simplified way of creating a directory structure for dumps, remove redundant methods in base called get_data_path and build_base_directory for non vert. Now get_dir is used by all modules except BLAST/BLAT since they need a different directory structure. get_dir has been extended to include division and check on species. Renamed config files to vertebrates and non_vertebrates to match other pipelines. This pipeline has been tested on sample databases from  all the divisions including human and collection database. Removed uncessary dump_fasta_dna dummy analysis and renamed CheckAssemblyGeneset

**Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion**

## Requirements

- Filling out the template is required. 
- Review the [contributing guidelines](https://github.com/Ensembl/ensembl/blob/master/CONTRIBUTING.md#why-could-my-pull-request-be-rejected) for this repository; remember in particular:
    - do not modify code without testing for regression
    - provide simple unit tests to test the changes
    - if you change the schema you must patch the test databases as well, see [Updating the schema](https://github.com/Ensembl/ensembl/blob/master/CONTRIBUTING.md#updating-the-schema)
    - the PR must not fail unit testing

## Description

Retired eg flag, eg-release, previous_release, eg_version and sub_dir from config files and associated modules. Simplified way of creating a directory structure for dumps, remove redundant methods in base called get_data_path and build_base_directory for non vert. Now get_dir is used by all modules except BLAST/BLAT since they need a different directory structure. get_dir has been extended to include division and check on species. Renamed config files to vertebrates and non_vertebrates to match other pipelines. This pipeline has been tested on sample databases from  all the divisions including human and collection database. Removed uncessary dump_fasta_dna dummy analysis and renamed CheckAssemblyGeneset

## Use case

This is mainly to clean up the pipeline and remove non-required hack since vertebrates now has a proper division.

## Benefits

Pipeline and modules a bit cleaner now, all using get_dir to create and retrieve dumps dir structure. Will make changes in the future easier. 

## Possible Drawbacks

None really.

## Testing

- [N ] Have you added/modified unit tests to test the changes?
- [ ] If so, do the tests pass?
- [ ] Have you run the entire test suite and no regression was detected?
- [ ] TravisCI passed on your branch

Dependencies
------------

Documentation will need to be updated.

### CODE REVIEW

File: modules/Bio/EnsEMBL/Production/Pipeline/GTF/DumpFile.pm [hunk 1]
```diff
@@ -70,39 +70,17 @@ sub param_defaults {
...
-     $self->param('release', $release);
-  } 
-  
+
   throw "Need a species" unless $self->param('species');
   throw "Need a release" unless $self->param('release');
   throw "Need a base_path" unless $self->param('base_path');
 
-  if(!$eg){
-   throw "No gtfToGenePred executable given"
```
Comment (james-monkeyshines) [inline modules/Bio/EnsEMBL/Production/Pipeline/GTF/DumpFile.pm]: Do we not still need to check for these files? Seems like they are used in vertebrate context.

File: modules/Bio/EnsEMBL/Production/Pipeline/GTF/DumpFile.pm [hunk 2]
```diff
@@ -167,7 +145,7 @@ sub run {
 
   if ($gene){
     $self->info(sprintf "Checking GTF file %s", $out_file);
-    $self->_gene_pred_check($out_file) unless $eg;
+    $self->_gene_pred_check($out_file) unless $self->division ne 'vertebrates';
```
Comment (james-monkeyshines) [inline modules/Bio/EnsEMBL/Production/Pipeline/GTF/DumpFile.pm]:
```suggestion
    $self->_gene_pred_check($out_file) if $self->division eq 'vertebrates';
```

File: modules/Bio/EnsEMBL/Production/Pipeline/GTF/DumpFile.pm (hunk 1 above)
Comment (thomasmaurel) [reply modules/Bio/EnsEMBL/Production/Pipeline/GTF/DumpFile.pm]: @james-monkeyshines: Thanks for spotting this. I think I was a bit overzealous when cleaning up. I think it's worth checking that these exists and are executable for vertebrates. I've pushed a new change.

File: modules/Bio/EnsEMBL/Production/Pipeline/GTF/DumpFile.pm (hunk 2 above)
Comment (thomasmaurel) [reply modules/Bio/EnsEMBL/Production/Pipeline/GTF/DumpFile.pm]: Added it in the commit to restore checks on GTF checks. Thanks for that.

**Pull request merged**